config.proto

```diff
@@ -127,3 +130,17 @@ message JavaAgent {
     // filter_jar_paths is the list of path to filter jars, separated by `,`
     repeated google.protobuf.StringValue filter_jar_paths = 1;
 }
+
+// CustomDataCaptureEndpoint represents a custom data capture rule for an endpoint that matches
+// the given url pattern and host header
+message CustomDataCaptureEndpoint {
```

Comment:
Not sure about calling it `Endpoint`. `Rule` is better I think? But I'm open to other names too.